corehq/ex-submodules/casexml/apps/case/signals.py

```diff
@@ -14,11 +13,13 @@ def rebuild_form_cases(sender, xform, *args, **kwargs):
         rebuild_case_from_forms(domain, case_id, detail)
 
 
-xform_archived.connect(rebuild_form_cases)
-xform_unarchived.connect(rebuild_form_cases)
+_connected = False
 
-# only when one or more cases are updated as the result of an xform submission
-# the contract of this signal is that you should modify the form and cases in
-# place but NOT save them. this is so that we can avoid multiple redundant writes
-# to the database in a row. we may want to revisit this if it creates problems.
-cases_received = Signal()  # providing args: xform, cases
+
+def connect_signals():
+    global _connected
+    if _connected:
+        return
```

Comment:
I don't think this is necessary. As far as I know `AppConfig.ready` will only be called once on startup if Django is being initialized properly.

Reply:
I believe that is true, but I'd prefer when modules can stand alone. This feels like one of those situations where code _should_ be used in one way, but its easy enough to make sure that isn't misused, that I'd prefer to go with that